shared/metrics/src/main/java/io/pravega/shared/MetricsNames.java

```diff
@@ -194,4 +202,24 @@ public static String nameFromContainer(String metric, int containerId) {
     public static String globalMetricName(String stringName) {
         return stringName + "_global";
     }
+
+    /**
+     * For some metrics such as OpStats, Pravega generates corresponding fail metrics automatically.
+     * This method provides a standard way to create name of fail metric for a given metric
+     *
+     * @param metricName - the metric name for which fail metric is created
+     * @return - the name of fail metric
+     */
+    public static String failMetricName(String metricName) {
+        if (Strings.isNullOrEmpty(metricName)) {
+            return metricName;
+        }
+        String[] tags = metricName.split("\\.");
+        if (tags.length >= 2 && Ints.tryParse(tags[tags.length - 1]) != null) {
+            tags[tags.length - 2] += "_fail";
+            return String.join(".", tags);
+        } else {
+            return metricName + "_fail";
```

Comment:
Are there cases that fall in this case?

Reply:
they are different cases: with / without the last segment as integer (such as container id)

Reply:
@RaulGracia It might be a good idea to add some unit test for this class. It's doing non-trivial things so it might be good to have those covered. Would you mind creating an issue for that?

Reply:
Unit tests created already.

Reply:
Ok, I didn't see that. Thanks for pointing out.